niworkflows/interfaces/nibabel.py

```diff
@@ -89,3 +89,68 @@ def _run_interface(self, runtime):
         maskimg.to_filename(self._results['out_mask'])
 
         return runtime
+
+
+class _FourToThreeInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='input 4d image')
+
+
+class _FourToThreeOutputSpec(TraitedSpec):
+    out_files = OutputMultiObject(File(exists=True),
+                                     desc='output list of 3d images')
+
+
+class FourToThree(SimpleInterface):
+    """Split a 4D dataset along the last dimension
+    into a series of 3D volumes."""
+
+    input_spec = _FourToThreeInputSpec
+    output_spec = _FourToThreeOutputSpec
+
+    def _run_interface(self, runtime):
+        filenii = nb.load(self.inputs.in_file)
+        if len(filenii.shape) != 4:
+            raise RuntimeError('Input image (%s) is not 4D.' % filenii)
```

Comment:
Perhaps just return if it is already 3D - only fail otherwise?

Reply:
What about a case where the image should be 4D, but a 3D image is supplied by mistake? if we went with the `return`, maybe we could still retain a user warning?

Reply:
Two issues: better check of dimensions, and `filenii` is not a path that the string interpolation can take in. 

I'd suggest this:
(assuming the nb.squeeze_image and the new input above are accepted)
```suggestion
        ndim = filenii.dataobj.ndim
        if ndim != 4:
            if self.inputs.accept_3D and ndim == 3:
                out_file = str(
                    Path(fname_presuffix(self.inputs.in_file, suffix=f"_idx-000")).absolute()
                )
                self._results['out_files'] = out_file
                filenii.to_filename(out_file)
                return runtime
            raise RuntimeError(f"Input image image is {ndim}D.")
```